bench/lib/TokenizerBench.re

```diff
@@ -17,17 +17,27 @@ let splitEverything = (_, _, _, _) => true;
 let splitNothing = (_, _, _, _) => false;
 
 let tokenizeWithSplits = () => {
-  let _ = Tokenizer.tokenize(~f=splitEverything, giantString);
+  let _: list(Tokenizer.TextRun.t) =
+    Tokenizer.tokenize(
+      ~f=splitEverything,
+      ~endIndex=BufferLine.lengthSlow(giantString),
+      giantString,
+    );
   ();
 };
 
 let tokenizeWithoutSplits = () => {
-  let _ = Tokenizer.tokenize(~f=splitNothing, giantString);
+  let _: list(Tokenizer.TextRun.t) =
+    Tokenizer.tokenize(
+      ~f=splitNothing,
+      ~endIndex=BufferLine.lengthSlow(giantString),
+      giantString,
+    );
   ();
 };
```

Comment:
These are actual partial application bugs prevented by the warning and type annotations. This code was nothing after `~endIndex` was added because this then resulted in a partial application.

You might want to double check the BufferLine benchmarks after fixing this.